app/models/contact.rb

```diff
@@ -0,0 +1,26 @@
+# == Schema Information
+#
+# Table name: contact
+#
+#  id          :bigint(8)        not null, primary key
+#  provider_id :integer          not null
+#  type        :text
+#  name        :text
+#  email       :text
+#  telephone   :text
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+
+class Contact < ApplicationRecord
+  belongs_to :provider
+
+  enum type: {
```

Comment:
Can we all this `contact_type` maybe? `type` is a special column in active record for single table inheritance.

Reply:
I'd prefer to just rename the `inheritance_column` since this seems like the right name to use for this value.

One test I use for this is to ask myself, if we were going to use STI, would the values in this column likely be the ones to decide the class name? And I think the answer here is yes. What do you think?

Reply:
Makes sense!